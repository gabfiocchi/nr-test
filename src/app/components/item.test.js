/* eslint-disable no-undef */
import Item from './item';

const item = new Item({
    version: 1,
    name: 'Dummy name',
    apdex: 100
});
document.body.innerHTML = item.template();

describe('Item component', () => {
    it('should have state props', () => {
        expect(item.state.version).toBe(1);
        expect(item.state.name).toBe('Dummy name');
        expect(item.state.apdex).toBe(100);
    });
    it('should render the template', () => {
        expect(item.render()).toMatchSnapshot();
    });
    it('should call to alert', () => {
        console.log('docu', document.getElementsByClassName('item')[0].innerHTML);
    });
});
