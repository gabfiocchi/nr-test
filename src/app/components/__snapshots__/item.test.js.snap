// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component should render the template 1`] = `
"
            <li
                class=\\"item\\"
                onClick=\\"showAlert(1)\\"
            >
                <span>100</span>
                <p>Dummy name</p>
            </li>
        "
`;

exports[`Item component should render the template 1`] = `
"
            <li
                class=\\"item\\"
                onClick=\\"showAlert(1)\\"
            >
                <span>100</span>
                <p>Dummy name</p>
            </li>
        "
`;
